resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -79,6 +79,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val useFetcherCache = conf.get(ENABLE_FETCHER_CACHE)
 
   private val maxGpus = conf.get(MAX_GPUS)
+  private val diskPerExecutor = conf.get(EXECUTOR_DISK)
```

Comment:
I think this throws an exception if not set, and it's optional. You can either check `.contains` here and make this an Option, or have a default value of '0' or something that would indicate no reservation.